hell/adventcalendar/2025/30/index.md

```diff
@@ -1,35 +1,200 @@
+* The last `<div>` was replaced with the [`<p>` element](https://html.spec.whatwg.org/multipage/grouping-content.html#the-p-element) – after all, it's a paragraph of text.
+
+If we open the document in the browser, it looks _better_:
+
+<img src="./html-article.png" alt="The &quot;Comandeer&quot; is now bigger and bolder, indicating that it is a heading and the &quot;Proud HTML Expert&quot; got a margin around itself.">
+
+In other words, we _formatted_ our text!
+
+<p class="highlight"><strong>Note:</strong> Each HTML element has its own <a href="https://html.spec.whatwg.org/multipage/rendering.html#rendering">default styles</a>, used to display it when there are no stylesheets provided by the website owner. This is the purest achievable form of HTML in a browser.</p>
+
+But it isn't particularly useful on its own. After all, there's always some CSS to make things prettier – why should we care about choosing appropriate HTML elements? Because the Web should not be only visual. One of the main principles of Web Content Accessibility Guidelines (WCAG) is [<q>Perceivable</q>](https://w3c.github.io/wcag/guidelines/22/#perceivable):
+
+> Information and user interface components must be presentable to users in ways they can perceive.
+
+In other words, if someone can't perceive visual information, it **must** be available for them in another form. A bunch of styled `<div>`s do not convey any meaning without their visuals. And that's problematic for assistive technology, like [screen readers](https://developer.mozilla.org/en-US/docs/Glossary/Screen_reader). Screen readers use an [accessibility tree](https://developer.mozilla.org/en-US/docs/Glossary/Accessibility_tree) – a DOM-like structure, built from the page's HTML and containing information about roles and purposes of all HTML elements. Let's see what that tree looks like for the `<div>` version of the page:
+
+<img src="./a11y-div.png" alt="Both &quot;Comandeer&quot; and &quot;Proud HTML Expert&quot; are marked up as static text.">
+
+It's just a text. More importantly, the image is missing, due to its empty `alt` attribute. The `<article>` version is much better:
+
+<img src="./a11y-article.png" alt="The whole bio is wrapped in the article, the &quot;Comandeer&quot; is correctly identified as a heading, the image has a description of &quot;A photo of smiling Comandeer&quot; and the &quot;Proud HTML Expert&quot; is marked up as a paragraph.">
+
+Thanks to that, the screen reader can successfully transform the page content into voice and inform the user where the articles, headings, and other elements are. It would not only allow the user to fully understand the page (similarly to how the non-screen-reader user perceived the visual version of the page) but also allow the screen reader to provide better ways of navigating the page, e.g., by [jumping to headings](https://webaim.org/projects/screenreadersurvey10/#finding).
```

Comment:
```suggestion
Thanks to that, the screen reader can successfully transform the page content into voice or braille and inform the user where the articles, headings, and other elements are. It would not only allow the user to fully understand the page (similarly to how the non-screen-reader user perceived the visual version of the page) but also allow the screen reader to provide better ways of navigating the page, e.g., by [jumping to headings](https://webaim.org/projects/screenreadersurvey10/#finding).
```